tests/test_cli.py

```diff
@@ -0,0 +1,140 @@
+from unittest import mock
+from io import StringIO
+import pytest
+
+import buku
+
+
+@pytest.fixture
+def stdin(monkeypatch):
+    with monkeypatch.context():
+        monkeypatch.setattr('sys.stdin', (buffer := StringIO()))
+        yield buffer
+
+@pytest.fixture
+def BukuDb():
+    with mock.patch('buku.BukuDb') as cls:
+        cls.return_value.close_quit.side_effect = SystemExit
+        yield cls
+
+@pytest.fixture
+def bdb(BukuDb):
+    yield BukuDb.return_value
+
+@pytest.fixture
+def piped_input():
+    with mock.patch('buku.piped_input') as fn:
+        yield fn
```

Comment:
Technically I could use `mock.patch()` as a decorator, but these are harder to name